flutter_maps_firestore/lib/main.dart

```diff
@@ -16,7 +17,11 @@ const _pinkHue = 350.0;
 // Places API client used for Place Photos
 final _placesApiClient = GoogleMapsPlaces(apiKey: googleMapsApiKey);
 
-void main() => runApp(App());
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
+  runApp(App());
+}
```

Comment:
If this is not following the coding standards, then I can also use a `FutureBuilder` to initialize. Let me know your thoughts.

Reply:
If you're referring to the `ensureInitialized` call, it's fine to do that in a sample. If this were a production app and we were focused on keeping launch time to the bare minimum it might be an issue, but code-readability is more important here.

Reply:
I've struggled with this one myself. I tend to put firebase initialisation closer to where it is used, eg in a provider state object or similar.

No hard guidelines